js/StickySidebarComponent.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+var $ = require('jquery');
+
+class StickySidebarComponent {
+
+    /**
+     * StickySidebarComponent
+     * @constructor 
+     * @param {object} window - window object
+     * @param {jQuery} $html - jQuery wrapper of the html node
+     */
+    constructor ($html, window) {
+        this.$html = $html;
+        this.window = window;
+    }
+
+    /**
+     * Initialise component
+     * 
+     * Should be fired on page load, checks for required options, performs an 
+     * initial sticky calculation and binds to resize & scroll events
+     */
+    init () {
+        if (typeof this.$html === 'undefined' || !this.$html) {
+            throw new Error('$html must be passed to StickySidebarComponent');
+        }
+
+        if (typeof this.window === 'undefined' || !this.window) {
+            throw new Error('window must be passed to StickySidebarComponent');
+        }
```

Comment:
we know that `this.$html` & `this.window` are declared, so these undefined checks should just be 
```
this.$html === undefined
this.window === undefined
```

also I think the `|| !this.$html` check is confusing, what are we asserting here exactly?

Reply:
I'm asserting that I copied it from StickyScrollbarComponent 😅

Reply:
haha! I suggest we remove it, as it is effectively asserting that if you pass in an argument that is falsy then throw an error (which is kind of fine), but what if I pass in an argument that is wrong but is truthy?